### DISCUSSION TITLE
Species Distribution Plot ValueError: failed to validate MultiPolygons....

### DISCUSSION THREAD

**Discussion by hanried** [earthlab-education/Earth-Analytics-AY24#562] [Q&A]
**What I am trying to do:** plot the veery migration in the species-distribution-33-plot notebook and plot my portfolio post species' (blue jay) migration.

**The code I'm running, the error I'm getting, and my attempted fix:**
CODE:
![image](https://github.com/user-attachments/assets/a36f6502-9b40-46ca-b4f2-7090f57a8682)

ERROR:
![image](https://github.com/user-attachments/assets/d43fafad-41a4-4806-9055-dbd31fe568b5)

The full language of the error is: "AbbreviatedException: ValueError: failed to validate MultiPolygons(id='p16154', ...).hatch_color: expected an element of either String, Nullable(Color), Instance(Value), Instance(Field), Instance(Expr), Struct(value=Nullable(Color), transform=Instance(Transform)), Struct(field=String, transform=Instance(Transform)) or Struct(expr=Instance(Expression), transform=Instance(Transform)), got dim('norm_occurrences')"

ATTEMPTED FIX:
Based on discussion in [this similar github issue](https://github.com/holoviz/hvplot/issues/649), I tried adding "rasterize=True" to my .hvplot code:
![image](https://github.com/user-attachments/assets/fa8d5de3-56d8-414d-b05c-b1f01ec8b870)

And I'm getting an interactive plot, however the edges of the ecoregions are really pixelated and don't have nice defined boundaries like the interactive plot on the textbook page has:
![image](https://github.com/user-attachments/assets/a22e0688-a1f4-4641-9f40-23465c6e6db4)

Any thoughts on:
1. The cause of the "AbbreviatedException: ValueError...", and
2. How to actually fix the "AbbreviatedException: ValueError..."

would be appreciated, thanks!!

**Comment by npwelsh** [comment]
Also getting the same issue when running my code.  It works when using the example data (tested again after this failed to make sure), but does not like the data I downloaded from GBIF.  
Hannah I tired your  rastering fix with my data, and it works pretty decent for me too, but in addition to the issues you had above the map is very slow and does not seem want to change with the slider. 

My next thought right now is that maybe the data downloaded from GBIF is too big (.csv is 159 MB vs 16.7 MB for the sample csv)  or needs to be formatted more?  I will play around more with this tomorrow.

**Reply by hanried** [reply]
I tried downloading the .html file of my interactive plot with the rastering fix. When I open the .html file in Chrome, the slider works smoother than it did in codespaces. So maybe you could try that to check at least whether the slider code is working by downloading the plot? Maybe it is slow/not working just because you're in your codespace still.

I hadn't thought about the data being too big, that's a good thought. Good luck tinkering with things today, I'll try to look at things this evening too and see if there's more formatting to be done with the gbif/ecoregion data to make things work smoother.

**Comment by eculler** [comment]
Hi @hanried @npwelsh -- @annaposlednik  had this problem as well. It seems to be a problem with the ecoregions GeoDataFrame, but we haven't gotten it sorted yet. However, a quick fix is to comment the line with `geo=True` -- the error seems to happen when the ecoregions get reprojected. I'd recommend going that route for now, probably.

If you do want to dig into this a bit more -- you could consider changing up your processing of the ecoregions GeoDataFrame. This has turned out to be a bit complicated with this dataset, especially if you want it to run quickly, but some things to try would be:

1. You can try reprojecting to mercator _before_ simplifying. I haven't tried that, but it seems to me like it might help avoid some of these errors.
2. You can try not using preserve_topology=False. This parameter speeds up the processing and helps avoid some errors, but probably also introduces some of these problems. I found that it was giving me errors, but only in the arctic/antarctic. So I selected my ecoregion data by REALM to avoid those errors and it worked ok. Honestly this was a bit of a rabbit hole for me.
3. If you are only looking at North America, the North American ecoregions from the EPA is a much more reliable file.
